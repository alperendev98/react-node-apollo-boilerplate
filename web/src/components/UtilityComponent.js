import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';
import AdminLayout from '../layouts/AdminLayout';
import TableContainer from '../containers/TableContainer';
import AddUtilityContainer from './UtilityComponents/AddUtilityContainer';

import '../styles/components/utilityComponent.css';

import addIcon from '../assets/images/icon-add.svg';

const BaseClasses = getBEMClasses(['base']);
const UtilityClasses = getBEMClasses(['utility']);

class UtilityComponent extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidUpdate() {}

	render() {
		const { showAddUtilityModal } = this.props;

		return (
			<div className={UtilityClasses('wrapper')}>
				<AdminLayout>
					<div className={UtilityClasses('container')}>
						<div className={UtilityClasses('header')}>
							<div className={UtilityClasses('header-title')}>Utillity Management</div>
							<div className={UtilityClasses('header-buttons')}>
								<div onClick={showAddUtilityModal} className={UtilityClasses('header-button--add')}>
									<div className={UtilityClasses('button-icon-label')}>Add Utility</div>
									<img src={addIcon} className={BaseClasses('icon')} />
								</div>
							</div>
						</div>
						<div className={UtilityClasses('content')}>
							<TableContainer />
						</div>
					</div>
				</AdminLayout>
				<AddUtilityContainer />
			</div>
		);
	}
}

export default UtilityComponent;
