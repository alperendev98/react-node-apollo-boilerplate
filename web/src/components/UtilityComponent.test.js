import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import UtilityComponent from './UtilityComponent';

const mockStore = configureStore();

jest.mock('../layouts/AdminLayout', () => {
	return () => <div className="admin-layout" />;
});

jest.mock('./UtilityComponents/AddUtilityContainer', () => {
	return () => <div className="add-utility" />;
});

describe('UtilityComponent', () => {
	const initialState = {};
	let component, wrapper, store;

	beforeEach(() => {
		store = mockStore(initialState);
		component = mount(<UtilityComponent store={store} />, {});
		wrapper = shallow(<UtilityComponent />);
	});

	it('shallowly renders UtilityComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('UtilityComponent should render AdminLayout', () => {
		expect(component.find('.admin-layout').length).toBe(1);
	});

	it('should match previos snapshot for UtilityComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
