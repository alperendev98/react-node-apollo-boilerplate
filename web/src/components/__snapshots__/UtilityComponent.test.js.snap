// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilityComponent should match previos snapshot for UtilityComponent 1`] = `
<div
  className="utility__wrapper"
>
  <Component>
    <div
      className="utility__container"
    >
      <div
        className="utility__header"
      >
        <div
          className="utility__header-title"
        >
          Utillity Management
        </div>
        <div
          className="utility__header-buttons"
        >
          <div
            className="utility__header-button--add"
          >
            <div
              className="utility__button-icon-label"
            >
              Add Utility
            </div>
            <img
              className="base__icon"
              src="icon-add.svg"
            />
          </div>
        </div>
      </div>
      <div
        className="utility__content"
      >
        <Connect(TableComponent) />
      </div>
    </div>
  </Component>
  <Component />
</div>
`;
